design/unnumbered-bgp.md

```diff
@@ -0,0 +1,436 @@
+kind-control-plane(eth00) 4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+kind-worker(eth01)        4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+kind-worker2(eth02)       4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+
+
+> k -n metallb-system exec -it -c frr frr-k8s-daemon-kzpz2 -- vtysh -c 'show bgp summary'
+
+IPv4 Unicast Summary (VRF default):
+BGP router identifier 172.19.0.3, local AS number 65000 vrf-id 0
+BGP table version 2
+RIB entries 3, using 288 bytes of memory
+Peers 1, using 13 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+unnumbered-p2p-peer(net0) 4      65004       512       512        2    0    0 00:25:15            1        2 TOR
+
+Total number of neighbors 1
+
+IPv6 Unicast Summary (VRF default):
+BGP router identifier 172.19.0.3, local AS number 65000 vrf-id 0
+BGP table version 1
+RIB entries 1, using 96 bytes of memory
+Peers 1, using 13 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+unnumbered-p2p-peer(net0) 4      65004       512       512        1    0    0 00:25:15            0        1 TOR
+
+Total number of neighbors 1
+```
+### Topics
+
+#### Remote-AS
+
+Strictly speaking, we do no need but is general consider part of the unnumbered
+[remote-as](https://docs.frrouting.org/en/latest/bgp.html#clicmd-neighbor-PEER-interface-remote-as-internal-external-auto-ASN)
+
+```
+neighbor PEER interface remote-as <internal|external|auto|ASN> Configure an
+unnumbered BGP peer. PEER should be an interface name. The session will be
+established via IPv6 link locals. Use internal for iBGP and external for eBGP
+sessions, or specify an ASN if you wish. Finally this connection type is meant
+for point to point connections. If you are on an ethernet segment and attempt
+to use this with more than one bgp neighbor, only one neighbor will come up,
+due to how this feature works.
+```
+
+#### Loopback IPv4 address
+
+Linux sends and receives IPv4 packets over an interface that does not have an IPv4 address. 
+
+#### RA enabled on both ends of the link
+
+The observation is that peering is being established even if RA are not enabled on both side.
+Docs are unclear if needed.
+
+#### BGP RouterID
+
+//TODO is there a requirement for that?
+
+### Traffic forwarding
+
+FRR advertises IPv4 prefixes with IPv6 next hops using encoding specified in RFC 5549 (now RFC 8950).
+
+### The API Extension
+
+#### frr-k8s side
+
+```go
+...
+
+type Neighbor struct {
+  Description string `json:"description"` // optional
+  Address string `json:"address"` // must be empty if interface != nil
```

Comment:
does this really need to be empty? could both being specified solve anything for https://github.com/metallb/metallb/issues/2423?